employee/admin.py

```diff
@@ -1,16 +1,161 @@
+# Remove fieldsets
+
+
+def remove_fieldsets(fieldsets, fieldsetname):
+    new_fieldsets = []
+    for fieldset in fieldsets:
+        if fieldset[0] is not fieldsetname:
+            new_fieldsets.append(fieldset)
+    return new_fieldsets
+
+
+class EmployeeAdmin(UserAdmin):
+    add_form = EmployeeCreationForm
+    form = EmployeeChangeForm
+    model = Employee
+    fieldsets = (
+        (None, {'fields': ('fullname', 'email', 'password')}),
+        ('Permissions', {'fields': ('is_staff',
+                                    'is_superuser', 'is_active')}),
+        ('Personal Information',
+            {
+                'fields':
+                (
+                    'dob',
+                    'gender',
+                    'phone_number',
+                    'address',
+                    'pan_no',
+                    'citizenship_no',
+                    'designation',
+                    'picture',
+                    'pan_no_document',
+                    'citizenship_document',
+                    'department',
+                )
+            }
+         ),
+    )
+    add_fieldsets = (
+        (None, {
+            'classes': ('wide',),
+            'fields': (
+                'fullname',
+                'email',
+                'password1',
+                'password2',
+                'dob',
+                'gender',
+                'phone_number',
+                'address',
+                'pan_no',
+                'citizenship_no',
+                'designation',
+                'picture',
+                'pan_no_document',
+                'citizenship_document',
+                'department',
+                'date_joined',
+            )
+        }
+        ),
+    )
+    exclude = ("created_by", "updated_by")
+    list_display = (
+        "picture_tag",
+        "fullname",
+        "date_joined",
+        "designation",
+        'department',
+        'is_active'
+    )
+    list_filter = ("designation", "department", "date_joined")
     search_fields = ["fullname"]
+    ordering = ('email',)
+
+    def get_queryset(self, request):
+        if request.user.is_superuser:
+            return Employee.objects.all()
+        return Employee.objects.filter(email=request.user.email)
+
+    def has_add_permission(self, request):
+        return request.user.is_superuser
+
+    def has_delete_permission(self, request, obj=None):
+        return request.user.is_superuser
+
+    def get_fieldsets(self, request, obj=None):
+        fieldsets = super(EmployeeAdmin, self).get_fieldsets(request, obj)
+        if not request.user.is_superuser and request.user.groups.filter(name='publisher').count() == 0:
+            return remove_fieldsets(fieldsets, 'Permissions')
+        return fieldsets
+
+    def get_form(self, request, obj=None, **kwargs):
+        form = super().get_form(request, obj, **kwargs)
+
+        disabled_fields = set()
+        if request.user.is_staff and not request.user.is_superuser:
```

Comment:
`is_staff` is not required inside the admin panel.
```suggestion
        if not request.user.is_superuser:
```

Reply:
Done as suggested and thanks.